File: 2418-sort-the-people/2418-sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        vector<pair<int,string>>ans;
        vector<string>res;
        for(int i=0;i<names.size();++i){
            ans.emplace_back(heights[i],names[i]);
        }
        sort(begin(ans),end(ans));
        reverse(begin(ans),end(ans));
        for(auto  [x,y]:ans){
            res.push_back(y);
        }
        return res;
    }
};

File: 2418-sort-the-people/2418-sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        temp = []
        for i in range(len(names)):
            temp.append([heights[i],names[i]])
        temp.sort()
        idx =0
        for i in reversed(range(len(names))):
            names[idx]= temp[i][1]
            idx+=1
        return names